### ISSUE TITLE
Multiple lines to navigation target

### ISSUE THREAD

**Issue by moving-bits** [cgeo/cgeo#13993]
### Describe your problem!

We have two users on support which experience multiple navigation lines being drawn to their current navigation target.

### How to reproduce?

One user gives the following description:
- Upgraded to latest system & now get multiple lines to cache when navigating via map. Seems to establish a line from the point where map nav was initiated and another from current location. Once yesterday I saw 4-5 lines from different points on my travel toward the cache.
- Here are pics I was able to capture today about the multiple vector lines.  I get two vector lines when switching from compass to map navigation. If I then switch to enable the live map, it generates multiple vector lines.

### Actual result after these steps?

![image](https://user-images.githubusercontent.com/3754370/219899608-937326d8-babf-451c-bcac-bbd65b7a8c05.png)

### Reproducible

Unclear

### c:geo Version

2023.02.12

### System information

```text
## System information

c:geo version: 2023.02.12

Device:
-------
- Device type: SM-S901U (r0qsqw, samsung)
- Available processors: 8
- Android version: 13
- Android build: TP1A.220624.014.S901USQS2BWA2
- Screen resolution: 1080x2115px (360x705dp)
- Pixel density: 3.0
- System font scale: 1.0 / used scale: 1.0
- Sailfish OS detected: false
- Google Play services: enabled - 23.04.13 (190408-505809224)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:113
- Set language: en_US (system default)
- System date format: M/d/yy
- Time zone: CST
- Debug mode active: no
- Last backup: never
- Routing mode: Walk
- Live map mode: true
- OSM multi-threading: false / threads: 1
- Map: Google: Satellite
- Id: cgeo.geocaching.maps.google.v2.GoogleMapProvider$GoogleSatelliteSource
- Atts: none
- Theme: none

Filters:
-------
- Hide waypoints: -
- LIVE: Cache Type: Adv Lab ([:inconclusive=false:advanced=false]AND(type:ADVLAB;difficulty_terrain:d=-:d=-:t=-:t=-;status))
- OFFLINE: None ([:inconclusive=false:advanced=false]AND(type;difficulty_terrain:d=-:d=-:t=-:t=-;status))

Stored Filters (#0):

Services:
-------
- Geocaching sites enabled:
geocaching.com: Logged in (Login OK) / PREMIUM
Geocaching.com Adventure Lab
- Geocaching.com date format: MM/dd/yyyy
- Geocaching.com website language: en-US
- Last login error on geocaching.com: communication error (2022-12-16 11-43)
- Last successful login on geocaching.com: 2023-02-17 10-25
- Routing: internal / BRouter installed: false
- Installed c:geo plugins: none

Permissions
-------
-android.permission.ACCESS_COARSE_LOCATION:: granted
-android.permission.ACCESS_FINE_LOCATION:: granted
-android.permission.READ_CONTACTS:: DENIED
-android.permission.READ_EXTERNAL_STORAGE:: DENIED

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (7.0 GB free) v3 internal isDir(8 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (7.0 GB free) v3 internal isDir(8 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (7.0 GB free) v3 external non-removable isDir(268 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (7.0 GB free) v3 internal isDir(0 entries)
- Public Folders: #12
- BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=41, dirs:>=9, totalFileSize:>=3.6 GB, free space: 7.0 GB, files on device: 13511677)
- OFFLINE_MAPS: /cgeo/maps (Default)[/cgeo/maps[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/maps]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fmaps, Av:true, files:2, dirs:2, totalFileSize:3.5 GB, free space: 7.0 GB, files on device: 13511677)
- OFFLINE_MAP_THEMES: /cgeo/maps/_themes (Default)[/cgeo/maps/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/maps/_themes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fmaps%2F_themes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 7.0 GB, files on device: 13511677)
- OFFLINE_MAP_SHADING: /cgeo/maps/_hgt (Default)[/cgeo/maps/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/maps/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fmaps%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 7.0 GB, files on device: 13511677)
- LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:4, dirs:0, totalFileSize:874.4 KB, free space: 7.0 GB, files on device: 13511677)
- GPX: /cgeo/gpx (Default)[/cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fgpx, Av:true, files:35, dirs:0, totalFileSize:14.0 MB, free space: 7.0 GB, files on device: 13511677)
- BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=8, dirs:>=6, totalFileSize:>=14.9 MB, free space: 7.0 GB, files on device: 13511677)
- FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 7.0 GB, files on device: 13511677)
- SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 7.0 GB, files on device: 13511677)
- ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:7, dirs:1, totalFileSize:81.2 KB, free space: 7.0 GB, files on device: 13511677)
- ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 7.0 GB, files on device: 13511677)
- TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 7.0 GB, files on device: -1)
- Map render theme path:
- PersistedDocumentUris: #1
- UNUSED_TRACK: null
- Persisted Uri Permissions: #1
- content://com.android.externalstorage.documents/tree/primary%3Acgeo (Sep 17, 2022, 16:17):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:6.3 MB) on system internal storage
```

### Additional Information

tickets 119963 (shown above) and 160607

**Comment by moving-bits** [comment]
Info from second user:

![image](https://user-images.githubusercontent.com/3754370/219899733-a3211c71-d6e7-41b2-91ad-fd7abaf54b86.png)

```
## System information

c:geo version: 2023.02.12

Device:
-------
- Device type: M2007J3SG (apollopro_eea, Xiaomi)
- Available processors: 8
- Android version: 12
- Android build: SKQ1.211006.001 test-keys
- Screen resolution: 1080x2180px (392x792dp)
- Pixel density: 2.8874998
- System font scale: 1.05 / used scale: 1.05
- Sailfish OS detected: false
- Google Play services: enabled - 23.04.13 (190400-505809224)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:183
- Set language: de_DE (system default)
- System date format: dd.MM.yy
- Time zone: GMT+01:00
- Debug mode active: no
- Last backup: 24. Okt. 2022, 21:54
- Routing mode: Straight
- Live map mode: true
- OSM multi-threading: false / threads: 1
- Map: Google: Satellit
  - Id: cgeo.geocaching.maps.google.v2.GoogleMapProvider$GoogleSatelliteSource
  - Atts: none
  - Theme: none

Filters:
-------
- Hide waypoints: -
- LIVE: Keine ([:inconclusive=true:advanced=true]AND(type;status))
- OFFLINE: Keine ([:inconclusive=false:advanced=true]AND(name::contains))

Stored Filters (#0):

Services:
-------
- Geocaching sites enabled:
   [geocaching.com](http://geocaching.com/): Logged in (Anmeldung OK) / PREMIUM
- [Geocaching.com](http://geocaching.com/) date format: dd/MM/yyyy
- [Geocaching.com](http://geocaching.com/) website language: en-US
- Last login error on [geocaching.com](http://geocaching.com/): communication error (2022-12-03 13-00)
- Last successful login on [geocaching.com](http://geocaching.com/): 2023-02-14 09-22
- Routing: internal / BRouter installed: false
- Installed c:geo plugins: contacts

Permissions
-------
-android.permission.ACCESS_COARSE_LOCATION:: granted
-android.permission.ACCESS_FINE_LOCATION:: granted
-android.permission.READ_CONTACTS:: DENIED
-android.permission.READ_EXTERNAL_STORAGE:: DENIED

Paths
-------
- System internal c:geo dir: /data/user/0/cgeo.geocaching (141,5 GB free) v3 internal isDir(8 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (141,5 GB free) v3 internal isDir(8 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (141,5 GB free) v3 external non-removable isDir(2941 entries)
- Internal theme sync (is turned ON): /data/user/0/cgeo.geocaching/MapThemeData (141,5 GB free) v3 internal isDir(14 entries)
- Public Folders: #12
  - BASE: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=490, dirs:>=16, totalFileSize:>=5,5 GB, free space: 141,5 GB, files on device: 12812797)
  - OFFLINE_MAPS: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=490, dirs:>=16, totalFileSize:>=5,5 GB, free space: 141,5 GB, files on device: 12812797)
  - OFFLINE_MAP_THEMES: /cgeo (User-Defined)[/cgeo[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo, Av:true, files:>=490, dirs:>=16, totalFileSize:>=5,5 GB, free space: 141,5 GB, files on device: 12812797)
  - OFFLINE_MAP_SHADING: /cgeo/_hgt (Default)[/cgeo/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 141,5 GB, files on device: 12812797)
  - LOGFILES: /cgeo/logfiles (Default)[/cgeo/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Flogfiles, Av:true, files:1, dirs:0, totalFileSize:11,8 KB, free space: 141,5 GB, files on device: 12812797)
  - GPX: /cgeo/gpx (Default)[/cgeo/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fgpx, Av:true, files:474, dirs:0, totalFileSize:499,8 MB, free space: 141,5 GB, files on device: 12812797)
  - BACKUP: /cgeo/backup (Default)[/cgeo/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Fbackup, Av:true, files:>=11, dirs:>=11, totalFileSize:>=859,9 MB, free space: 141,5 GB, files on device: 12812797)
  - FIELD_NOTES: /cgeo/field-notes (Default)[/cgeo/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 141,5 GB, files on device: 12812797)
  - SPOILER_IMAGES: /cgeo/GeocachePhotos (Default)[/cgeo/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 141,5 GB, files on device: 12812797)
  - ROUTING_BASE: /cgeo/routing (Default)[/cgeo/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting, Av:true, files:8, dirs:1, totalFileSize:93,6 MB, free space: 141,5 GB, files on device: 12812797)
  - ROUTING_TILES: /cgeo/routing/segments4 (Default)[/cgeo/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3Acgeo::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3Acgeo/document/primary%3Acgeo%2Frouting%2Fsegments4, Av:true, files:1, dirs:0, totalFileSize:93,5 MB, free space: 141,5 GB, files on device: 12812797)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 141,5 GB, files on device: -1)
- Map render theme path:
- PersistedDocumentUris: #1
- UNUSED_TRACK: null
- Persisted Uri Permissions: #1
  - content://com.android.externalstorage.documents/tree/primary%3Acgeo (27. Sept. 2022, 20:08):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:312,5 MB) on system internal storage

--- End of system information ---

```

**Comment by moving-bits** [comment]
Both users have provided screenshots for Google Maps Satellite only. I'll ask both whether OSM is affected as well. (No reports for OSM so far on support.)

**Comment by moving-bits** [comment]
Answer from user 2:
- He ist using GM only, will test with OSM on next occasion.
- One observation: Line seems to be redrawn on every GPS signal. You get a very regular pattern when driving with constant speed

![image](https://user-images.githubusercontent.com/3754370/221378796-9bb1b9b6-3522-4943-b4e4-67e1c928686c.png)

**Comment by eddiemuc** [comment]
Clarification questions:
* Is it correct that user would expect ONE line from his current position to navigation target? (e.g. user is in navigation mode and uses "direct" navigation via "Luftlinie")
* Are the lines deleted if user leaves map and then reenters it?

**Comment by moving-bits** [comment]
Feedback from user 2:
- Did not happen with OSM.
- Having lines on GM, then opening OSM, the extra lines are gone.
- Have asked him about closing / reopening GM behavior => waiting for feedback.

**Comment by moving-bits** [comment]
Feedback from user 2:
- Switching from GM to OSM removes the extra lines.
- No extra lines will appear while being in OSM mode.
- When switching back to GM, there are no extra lines visible in the beginning. => Have asked him whether there was no line at all after switching back to GM, and/or whether the extra lines showed up later.

**Comment by Lineflyer** [comment]
FYI: 
Another user on support with that problem.#
I advised to use OSM as a workaround.
If there are any questions to that user, ticket number is: 945212

**Comment by moving-bits** [comment]
Same for ticket 320972

**Comment by MagpieFourtyTwo** [comment]
Navigated with GSat today (curiosity kills the cat ;) ).
I mainly used "Direct Line" as in this mode it's pretty easy to see when a new line is drawn resp a zombie is left over.

My observations: Zombie lines can reproducible be forced by
- change of routing mode
- change of zoom level
- significant map movement (perhaps connected to map tile or data reload?)
- when switching to another app and back to c:geo, but only if the position changed significantly in the meantime - thus may in fact be the same as previous point

While not touching the phone (with activated "stay with me" button), zombies appeared just now and then, for maybe the same reason as #3 and 4 above. 

Perhaps even useful for #14068 and #14032

**Comment by eddiemuc** [comment]
I created and merged PR #14083 recently (on `release`), did this fix help mitigate the issue?

**Comment by fm-sys** [comment]
> I created and merged PR #14083 recently (on `release`), did this fix help mitigate the issue?

As branches were not merged back to `master` it was not yet included in the nightlys. I've just triggered our branch-merge workflow, I'll report back in the next few days...

**Comment by MagpieFourtyTwo** [comment]
WRT my points:
The survival of routing lines on **mode changes** appears to be solved with [5a5cfe1](https://github.com/cgeo/cgeo/commit/5a5cfe11d72fe3dc3a67c6f169b9d60238412ddc) resp. #14083, as tested with today's nightly.
Additionally, I successfully tested **zoom level changes** and **map movements** in GMap and GSat, but just in a simulation,
thus a RL test is still TBD.

Nevertheless, things look really good, IMO.

**Comment by fm-sys** [comment]
Just opened GMap while sitting in a train. Old lines are correctly removed again, no ghost lines anymore...
